docs/user/FeatureFiltering.asciidoc

```diff
@@ -0,0 +1,251 @@
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "amenity=hospital",
+      "allowAliases": "true"
+    }
+  ]
+}
+-----
+
+5) Filter down to restaurants and anything with a schema similarity to restaurants >= 0.8 (similarity comparison off by default).  Feature type similarity scores are controlled from the hoot schema.  For example, this also picks up _amenity=pub_:
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "amenity=restaurant",
+      "similarityThreshold": "0.8"
+    }
+  ]
+}
+-----
+
+6) Filter down to only non-building malls (POI's):
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "shop=mall"
+    }
+  ],
+  "must_not":
+  [
+    {
+      "tag": "building=*"
+    }
+  ]
+}
+-----
+
+7) Filter down to features with a tag key containing "address":
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "*address*=*"
+    }
+  ]
+}
+-----
+
+8) Filter down to features with a tag key starting with "address":
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "address*=*"
+    }
+  ]
+}
+-----
+
+9) Filter down to features with a tag key ending with "address":
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "*address=*"
+    }
+  ]
+}
+-----
+
+10) Filter down to features with a tag value containing "address":
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "*=*address*"
+    }
+  ]
+}
+-----
+
+11) Filter down to features with a tag value starting with "address":
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "*=address*"
+    }
+  ]
+}
+-----
+
+12) Filter down to features with a tag value ending with "address":
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "*=*address"
+    }
+  ]
+}
+-----
+
+13) Filter down to all gravel roads, as well as their descendants (off by default; this also returns surface=fine_gravel and surface=pebblestone):
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "surface=gravel",
+      "allowChildren": "true"
+    }
+  ]
+}
+-----
+
+14) Filter down to all roads even though highway=secondary was specified (off by default; this also returns highway=road):
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "highway=secondary",
+      "allowAncestors": "true"
+    }
+  ]
+}
+-----
+
+15) Query for all transportation related features (no tag filter may be specified with a category; current available categories include: poi, building, transportation, use, multiuse, name, and pseudoname):
+-----
+{
+  "must": 
+  [
+    {
+      "category": "transportation"
+    }
+  ]
+}
+-----
+
+16) Query for all features associated with building:part=yes (this is kind of catch all where other relationships are too strong of a link; associations aren't widely used in the hoot schema but can be added quite easily):
+-----
+{
+  "must": 
+  [
+    {
+      "tag": "building:part=yes",
+      "allowAssociations": "true"
+    }
+  ]
+}
+-----
+
```

Comment:
These are good explanations and great examples.

Reply:
+1